### PR TITLE
Location restore file restriction - other backend logic

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

JIra: [USH-4349](https://dimagi.atlassian.net/browse/USH-4349?atlOrigin=eyJpIjoiNTIyNjM0MWI4YzY3NDEwMmIwYzE0M2NlYjczYjI2YWYiLCJwIjoiaiJ9).

More backend logic for the "Location restore file restriction" feature. This PR is dependent on #34358.

## Technical Summary

`get_location_fixture_ids` gets the list of locations for a user restricted by the "restrict cases to" setting, but includes the current and ancestor locations. So the proper set of location owner IDs for the restore file should be the intersection of `get_location_fixture_ids` and `get_queryset_descendants`.

## Feature Flag

USH_LOCATION_RESTORE_FILE_CASE_SYNC_RESTRICTION, but this is to be GA'd soon.

### Safety story

- existing automated tests ensure no regression
- added new tests for new functionality that is behind a FF
- this will go through QA

### Automated test coverage

A few places: [1](https://github.com/dimagi/commcare-hq/blob/4edd9ecff61b07e40442f969fa0ad9584616bc41/corehq/apps/locations/tests/test_location_types.py#L100-L142), [2](https://github.com/dimagi/commcare-hq/blob/4edd9ecff61b07e40442f969fa0ad9584616bc41/corehq/apps/locations/tests/test_location_groups.py#L120-L138), and [3](https://github.com/dimagi/commcare-hq/blob/4edd9ecff61b07e40442f969fa0ad9584616bc41/corehq/apps/users/tests/test_get_owner_ids.py#L8).

### QA Plan

To be started soon

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-4349]: https://dimagi.atlassian.net/browse/USH-4349?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MartinRiese — APPROVED** [review]
If I got that right, this is only for the last commit.

**Comment by esoergel** [comment]
just noting that you can also set the base branch for this PR to the other one.  That'll make the diff limited to only those changes.  I think github now automatically switches the base branch back to `master` after the other PR is merged (if this one isn't merged first)

File: corehq/apps/locations/sql_templates/get_location_fixture_ids_2.sql
Comment (millerdev) [inline corehq/apps/locations/sql_templates/get_location_fixture_ids_2.sql:1]: Was this file intended to be added as part of a "fix spelling" commit?
